Seacas: new seacas build error in Albany nightlies

We have a new seacas build error in some of our Trilinos nightlies: 
```
Trilinos/packages/seacas/libraries/ioss/src/exodus/Ioex_IOFactory.C:121: undefined reference to `Ioex::ParallelDatabaseIO::ParallelDatabaseIO(Ioss::Region*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Ioss::DatabaseUsage, ompi_communicator_t*, Ioss::PropertyManager const&)'
```
https://sems-cdash-son.sandia.gov/cdash/viewBuildError.php?buildid=1803

I'm guessing it crept in with PR https://github.com/trilinos/Trilinos/pull/7368 ?  @gsjaardema , can you please have a look?

@trilinos/seacas

@gsjaardema : thanks for putting this together so fast!  I just tried building Albany with this PR and unfortunately other issues have cropped up, e.g.: 
```
[ 55%] Linking CXX executable io_info
/usr/bin/ld: ../exodus/libIoex.so.12.17: undefined reference to `Ioex::ParallelDatabaseIO::ParallelDatabaseIO(Ioss::Region*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Ioss::DatabaseUsage, ompi_communicator_t*, Ioss::PropertyManager const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [packages/seacas/libraries/ioss/src/main/CMakeFiles/io_info.dir/build.make:123: packages/seacas/libraries/ioss/src/main/io_info] Error 1
make[1]: *** [CMakeFiles/Makefile2:11650: packages/seacas/libraries/ioss/src/main/CMakeFiles/io_info.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
I am attaching the configure script I use to build the code.  You can probably reproduce the error if you use it.  

[do-cmake-trilinos-mpi-camobap.txt](https://github.com/trilinos/Trilinos/files/4637439/do-cmake-trilinos-mpi-camobap.txt)

--- a/packages/seacas/libraries/exodus/include/exodusII.h
+++ b/packages/seacas/libraries/exodus/include/exodusII.h
@@ -50,7 +50,6 @@
 #include "netcdf_meta.h"
 #if NC_HAS_PARALLEL
 #ifndef PARALLEL_AWARE_EXODUS
-#define PARALLEL_AWARE_EXODUS
 #endif
 #endif
 #endif

--- a/packages/seacas/libraries/ioss/src/exodus/Ioex_IOFactory.C
+++ b/packages/seacas/libraries/ioss/src/exodus/Ioex_IOFactory.C
@@ -33,7 +33,7 @@
 #include <exodus/Ioex_IOFactory.h> // for Ioex IOFactory
 #include <exodus/Ioex_DatabaseIO.h> // for Ioex DatabaseIO
 
-#if defined(SEACAS_HAVE_MPI)
+#if defined(PARALLEL_AWARE_EXODUS)
 #include <exodus/Ioex_ParallelDatabaseIO.h> // for Ioex ParallelDatabaseIO
 #endif
 #include <tokenize.h>
@@ -54,7 +54,7 @@ namespace Ioss {
   class DatabaseIO;
 } // namespace Ioss
 
-#if defined(SEACAS_HAVE_MPI)
+#if defined(PARALLEL_AWARE_EXODUS)
 namespace {
   std::string check_decomposition_property(const Ioss::PropertyManager &properties,
                                            Ioss::DatabaseUsage          db_usage);
@@ -76,7 +76,7 @@ namespace Ioex {
     Ioss::IOFactory::alias("exodus", "exodusii");
     Ioss::IOFactory::alias("exodus", "exodusII");
     Ioss::IOFactory::alias("exodus", "genesis");
-#if defined(SEACAS_HAVE_MPI)
+#if defined(PARALLEL_AWARE_EXODUS)
     Ioss::IOFactory::alias("exodus", "dof_exodus");
     Ioss::IOFactory::alias("exodus", "dof");
 #endif
@@ -86,7 +86,7 @@ namespace Ioex {
                                        MPI_Comm                     communicator,
                                        const Ioss::PropertyManager &properties) const
   {
-#if defined(SEACAS_HAVE_MPI)
+#if defined(PARALLEL_AWARE_EXODUS)
     // The "exodus" and "parallel_exodus" databases can both be accessed
     // from this factory.  The "parallel_exodus" is returned only if the following
     // are true:
@@ -142,7 +142,7 @@ namespace Ioex {
   }
 } // namespace Ioex
 
-#if defined(SEACAS_HAVE_MPI)
+#if defined(PARALLEL_AWARE_EXODUS)
 namespace {
   std::string check_decomposition_property(const Ioss::PropertyManager &properties,
                                            Ioss::DatabaseUsage          db_usage)

